main.go

```diff
@@ -0,0 +1,543 @@
+		// run the most important task we find, and then return, ignorning
+		// remaining urls for lower priority tasks that might still be left to
+		// loop through, since by the time we complete the first task, maybe
+		// higher priority jobs are waiting, so we need to poll afresh.
+		log.Println("Task found")
+		taskFound = true
+		// If there is one or more messages the worker must claim the tasks
+		// referenced in the messages, and delete the messages.
+		err = task.claim()
+		if err != nil {
+			log.Printf("WARN: Not able to claim task %v\n", task.TaskId)
+			log.Println(err)
+			continue
+		}
+		task.setReclaimTimer()
+		err = task.fetchTaskDefinition()
+		if err != nil {
+			log.Printf("WARN: Not able to fetch task definition for task %v\n", task.TaskId)
+			log.Println(err)
+			continue
+		}
+		err = task.validatePayload()
+		if err != nil {
+			log.Printf("WARN: Not able to validate task payload for task %v\n", task.TaskId)
+			log.Println(err)
+			continue
+		}
+		err = task.run()
+		if err != nil {
+			log.Printf("WARN: Not able to run task %v\n", task.TaskId)
+			log.Println(err)
+			continue
+		}
+		break
+	}
+	return taskFound
+}
+
+// Queries the given Azure Queue signed url pair (poll url/delete url) and
+// translates the Azure response into a Task object
+func (urlPair SignedURLPair) Poll() (*TaskRun, error) {
+	queueMessagesList := new(QueueMessagesList)
+	// To poll an Azure Queue the worker must do a `GET` request to the
+	// `signedPollUrl` from the object, representing the Azure queue. To
+	// receive multiple messages at once the parameter `&numofmessages=N`
+	// may be appended to `signedPollUrl`. The parameter `N` is the
+	// maximum number of messages desired, `N` can be up to 32.
+	// Since we can only process one task at a time, grab only one.
+	resp, err := http.Get(urlPair.SignedPollUrl + "&numofmessages=1")
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	// When executing a `GET` request to `signedPollUrl` from an Azure queue object,
+	// the request will return an XML document on the form:
+	//
+	// ```xml
+	// <QueueMessagesList>
+	//     <QueueMessage>
+	//       <MessageId>...</MessageId>
+	//       <InsertionTime>...</InsertionTime>
+	//       <ExpirationTime>...</ExpirationTime>
+	//       <PopReceipt>...</PopReceipt>
+	//       <TimeNextVisible>...</TimeNextVisible>
+	//       <DequeueCount>...</DequeueCount>
+	//       <MessageText>...</MessageText>
+	//     </QueueMessage>
+	//     ...
+	// </QueueMessagesList>
+	// ```
+	// We unmarshal the response into go objects, using the go xml decoder.
+	fullBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	reader := strings.NewReader(string(fullBody))
+	dec := xml.NewDecoder(reader)
+	err = dec.Decode(&queueMessagesList)
+	if err != nil {
+		return nil, err
+	}
+	if len(queueMessagesList.QueueMessages) == 0 {
+		return nil, errors.New("Zero tasks returned in Azure XML QueueMessagesList")
```

Comment:
This is not really an error... I would just return `nil, nil`... and treat as no message available... it's an expected case, using the error code path can be unpleasant...